std/uuid.d

```diff
@@ -1085,8 +1088,14 @@ unittest
     return u;
 }
 
+private @trusted
+ref T unsafeTrustingArrayElementReinterpret(T)(ref ubyte[16] arr, size_t i)
+{
+    return *cast(T*)(arr.ptr+i);
```

Comment:
Yes and no. It's private, meaning that whoever calls it has the responsibility of checking it's actual trustedness (as its name indicates). No matter how you look at it, someone somewhere will have to make it trusted. I'm not a huge fans of trusted lambdas either, and I'm not up-to-date on the entire inline issue either, so I trust this the most... So yes, it's abusive, but none more so that trusted lambdas, IMO.

Alternatives include giving UUID trusted (but guaranteed safe) members `ref T[16/T.sizeof] asArrayOf(T)()`, where T is integral?

Either that, or do some bit manipulation to do byte by byte copy, but I'm not a huge fan of giving up on efficiency when safety can argualy be proven.